packages/sdk/src/ipc/gateway.ts

```diff
@@ -74,55 +69,50 @@ function releaseParamsToTyped(address: Address): ReleaseParams {
 // Gateway manager facet for managing gateway funding and releasing
 export class GatewayManager {
   getContract(
-    client: RecallClient,
-    contractAddress?: Address,
+    publicClient: PublicClient,
+    walletClient: WalletClient,
+    contractAddress: Address,
```

Comment:
Note: this pattern is probably needed in the `entities` (buckets, blobs, etc) since `RecallClient` instantiates each entity, but then the entity itself takes a `RecallClient` in its constructor. It's self-referencing and not ideal, but I wanted to keep the PR small and can handle that later. will handle in #79